sdk/compiler/daml-lf-tools/src/DA/Daml/LF/InferSerializability.hs

```diff
@@ -15,21 +15,25 @@ import DA.Daml.LF.Ast
 import DA.Daml.LF.TypeChecker.Serializability (CurrentModule(..), serializabilityConditionsDataType)
 
 inferModule :: World -> Version -> Module -> Either String Module
-inferModule world0 version mod0 = do
-  let modName = moduleName mod0
-  let dataTypes = moduleDataTypes mod0
-  let interfaces = NM.namesSet (moduleInterfaces mod0)
-  let eqs =
-        [ (dataTypeCon dataType, serializable, deps)
-        | dataType <- NM.toList dataTypes
-        , let (serializable, deps) =
-                case serializabilityConditionsDataType world0 version (Just $ CurrentModule modName interfaces) dataType of
-                  Left _ -> (False, [])
-                  Right deps0 -> (True, HS.toList deps0)
-        ]
-  case leastFixedPointBy (&&) eqs of
-    Left name -> throwError ("Reference to unknown data type: " ++ show name)
-    Right serializabilities -> do
-      let updateDataType dataType =
-            dataType{dataSerializable = IsSerializable (HMS.lookupDefault False (dataTypeCon dataType) serializabilities)}
-      pure mod0{moduleDataTypes = NM.map updateDataType dataTypes}
+inferModule world0 version mod0 =
+  case moduleName mod0 of
+    -- non stable part of stdlib/prim cannot contain serializable type
```

Comment:
```suggestion
    -- Unstable parts of stdlib mustn't contain serializable types, because if they are 
    -- serializable, then the upgrading checks run on the datatypes and this causes problems. 
    -- Therefore, we mark the datatypes as not-serializable, so that upgrades checks don't trigger.
    -- For more information on this issue, refer to issue 
    -- https://github.com/digital-asset/daml/issues/19338issues/19338
```

Reply:
Looks great, thank you!